internal/connector/connector.go

```diff
@@ -151,7 +160,20 @@ func Run(ctx context.Context, options Options) error {
 			logging.Errorf("Unauthorized error; token invalid or expired. exiting.")
 			cancel()
 		},
-	}
+		ObserveFunc: func(start time.Time, request *http.Request, response *http.Response) {
+			statusCode := http.StatusInternalServerError
+			if response != nil {
```

Comment:
If the response is nil does that mean the request timed out?

Reply:
I think so though 500 Internal Server Error should be a good enough default to capture cases like this.